### PR TITLE
Add response metrics to the square backend

### PR DESCRIPTION
**Pull request by Tabrizian**

### CODE REVIEW

File: src/square.cc [hunk 1]
```diff
@@ -389,6 +401,14 @@ ModelInstanceState::RequestThread(
         factory.get(),
         TRITONBACKEND_ResponseNewFromFactory(&response, factory.get()));
 
+    uint64_t response_start;
+    SET_TIMESTAMP(response_start);
+
+    // In this specific example, there is not computation so the infer time is
```
Comment (tanmayv25) [inline src/square.cc]: no computation

File: src/square.cc [hunk 2]
```diff
@@ -380,6 +380,18 @@ ModelInstanceState::RequestThread(
       TRITONBACKEND_ResponseFactory, backend::ResponseFactoryDeleter>
       factory(factory_ptr);
 
+  if (element_count == 0) {
```
Comment (tanmayv25) [inline src/square.cc]: Why did we have to move this block on top? 
Shouldn't the logic be skipped for `element_count==0`? Sending such flags looks cleaner towards the end.

**Review by tanmayv25 — CHANGES_REQUESTED** [review]

File: src/square.cc (hunk 1 above)
Comment (kthui) [reply src/square.cc]: Updated.

File: src/square.cc (hunk 2 above)
Comment (kthui) [reply src/square.cc]: Updated.

**Review by tanmayv25 — APPROVED** [review]

**Pull request merged**